src/slices/GiveThunk.ts

```diff
@@ -0,0 +1,322 @@
+
+interface IUAData {
+  address: string;
+  value: string;
+  recipient: string;
+  approved: boolean;
+  txHash: string | null;
+  type: string | null;
+}
+
+export const PENDING_TXN_GIVE = "giving";
+export const PENDING_TXN_EDIT_GIVE = "editingGive";
+export const PENDING_TXN_WITHDRAW = "endingGive";
+export const PENDING_TXN_GIVE_APPROVAL = "approve_giving";
+
+export const ACTION_GIVE = "give";
+export const ACTION_GIVE_EDIT = "editGive";
+export const ACTION_GIVE_WITHDRAW = "endGive";
+
+export const isSupportedChain = (chainID: number): boolean => {
+  // Give is only supported on Ethereum mainnet (1) and rinkeby (4) for the moment.
+  if (chainID === 1 || chainID === 4) return true;
+
+  return false;
+};
+
+export const hasPendingGiveTxn = (pendingTransactions: IPendingTxn[]): boolean => {
+  return (
+    isPendingTxn(pendingTransactions, PENDING_TXN_GIVE) ||
+    isPendingTxn(pendingTransactions, PENDING_TXN_EDIT_GIVE) ||
+    isPendingTxn(pendingTransactions, PENDING_TXN_WITHDRAW)
+  );
+};
+
+// This is approving the recipient to spend, not the contract
+export const changeApproval = createAsyncThunk(
+  "give/changeApproval",
+  async ({ token, provider, address, networkID }: IChangeApprovalAsyncThunk, { dispatch }) => {
+    if (!provider) {
+      dispatch(error(t`Please connect your wallet`));
+      return;
+    }
+
+    /*
+      On testnet it's been best for testing Give to use a pseudo-sOHM contract
+      that gives us more control to rebase manually when needed. However, this 
+      makes it not as perfectly translatable to mainnet without changing any parameters
+      this is the best way to avoid manually switching out code every deployment
+    */
+    const signer = provider.getSigner();
+    const sohmContract = new ethers.Contract(addresses[networkID].SOHM_V2 as string, ierc20Abi, signer);
+    let approveTx;
+    try {
+      approveTx = await sohmContract.approve(
+        addresses[networkID].GIVING_ADDRESS,
+        ethers.utils.parseUnits("1000000000", "gwei").toString(),
+      );
+      const text = "Approve giving";
+      const pendingTxnType = PENDING_TXN_GIVE_APPROVAL;
+      dispatch(fetchPendingTxns({ txnHash: approveTx.hash, text, type: pendingTxnType }));
+      await approveTx.wait();
+    } catch (e: unknown) {
+      dispatch(error((e as IJsonRPCError).message));
+      return;
+    } finally {
+      if (approveTx) {
+        dispatch(clearPendingTxn(approveTx.hash));
+      }
+    }
+
+    /*
+      The pseudo-sOHM contract used on testnet does not have a functional allowance
+      mapping. Instead approval calls write allowances to a mapping title _allowedValue
+    */
+    let giveAllowance = await sohmContract.allowance(address, addresses[networkID].GIVING_ADDRESS);
+
+    return dispatch(
+      fetchAccountSuccess({
+        giving: {
+          sohmGive: +giveAllowance,
+        },
+      }),
+    );
+  },
+);
+
+export const changeMockApproval = createAsyncThunk(
```

Comment:
do you want to kill this now? Or will it still be necessary in the future?

Reply:
We want to keep this for testing purposes. Keep it.